lib/Doctrine/ORM/Query/Parser.php

```diff
@@ -302,7 +302,7 @@ public function getAST()
      */
     public function match($token)
     {
-        $lookaheadType = $this->lexer->lookahead['type'];
+        $lookaheadType = $this->lexer->lookahead ? $this->lexer->lookahead['type'] : null;
```

Comment:
Maybe using `??` would be a bit more cleaner?

Reply:
True. I didn't now that this operator would also handle attempting to access array offset on null without erroring. Fixed.